### PR TITLE
Add branching priority

### PR DESCRIPTION
**Pull request by laradicp**

### CODE REVIEW

File: src/branchingpriority.jl [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+function MOI.set(
+    dest::MOIU.UniversalFallback, attribute::VarBranchingPriority, value
+)
+    dest.modattr[attribute] = value
+    return
+end
+
+function MOI.get(dest::MOIU.UniversalFallback, attribute::VarBranchingPriority)
+    return get(dest.modattr, attribute, nothing)
```
Comment (guimarqu) [inline src/branchingpriority.jl:17]: I think the default priority should be 1. So the method should return 1 by default.

File: src/branchingpriority.jl [hunk 2]
```diff
@@ -0,0 +1,18 @@
+struct VarBranchingPriority <: MOI.AbstractVariableAttribute end
+
+branchingpriority!(model, x::JuMP.VariableRef, branching_priority::Int) = MOI.set(
```
Comment (guimarqu) [inline src/branchingpriority.jl:3]: Does it work if one wants to set the branching priority of a collection of variables ?

For instance : 
```julia
@variable(m, x[1:10])
branchingpriority!(x, 2)
```

**Review by guimarqu — COMMENTED** [review]
You should add some tests

File: src/branchingpriority.jl (hunk 2 above)
Comment (laradicp) [reply src/branchingpriority.jl:3]: Not yet, but it's simple to make that happen. Should `branchingpriority(model, x)` ("get" method) return a single `Int`?

Comment (guimarqu) [reply src/branchingpriority.jl:3]: Good question. I think we'll have different branching priorities in a given collection. Perhaps, the get method should have a behaviour similar to `JuMP.value` ?

File: test/branchingpriority.jl [hunk 3]
```diff
@@ -0,0 +1,17 @@
+function test_branching_priority()
+    model = Model()
+    @variable(model, x)
+    @variable(model, y[1:5])
+
```
Comment (guimarqu) [inline test/branchingpriority.jl:5]: Does it work for  

```julia
@variable(model, z[['a', 'b', 'd', 'e'], 1:5]) # Dense axis array
@variable(model, w[i in 1:10, j in 1:i]) # Sparse axis array
```

?

**Review by guimarqu — APPROVED** [review]

**Pull request merged**